api_bridge_agent_demos/mcp/README.md

```diff
@@ -0,0 +1,111 @@
+# API Bridge Agent MCP demo
+
+This application demonstrates the use of MCP (Model Context Protocol) in API
+Bridge Agent.
+
+The role of this application is to take a Github repository as input, analyze
+the repository and then look for related contents on DuckDuckGo search engine.
+
+The setup consists of:
+- An application developed with the LlamaIndex framework.
+- API Bridge Agent configured with at least the Github API, and the DuckDuckGo
+MCP server.
+
+# Configuration
+
+## API Bridge agent configuration
+
+Use [this MCP configuration file](https://github.com/agntcy/api-bridge-agnt/blob/main/configs/mcp.oas.json) as a starting point.
+
+Edit the file to add the DuckDuckGo MCP server:
+
+```json
+"mcpServers": {
+  "ddg": {
+    "command": "uvx",
+    "args": ["duckduckgo-mcp-server"]
+  }
+}
+```
+
+Start API Bridge Agent, configure it with MCP, in initialize the MCP servers:
+
+```shell
+curl http://localhost:8080/tyk/apis/oas \
+  --header 'x-tyk-authorization: foo' \
+  --header 'Content-Type: text/plain' \
+  -d@configs/mcp.oas.json
+
+curl http://localhost:8080/tyk/reload/group --header 'x-tyk-authorization: foo'
+
+curl http://localhost:8080/mcp/init
+```
+
+# Running the application locally
+
+```shell
+$ uv run demo --help
+Usage: demo [OPTIONS]
+
+Options:
+  --repository TEXT      [required]
+  --api-bridge-url TEXT
+  --help                 Show this message and exit.
+```
+
+Check for pages that are related to the 'agntcy/acp-spec' Github repository:
+
+```shell
+uv run demo --repository 'agntcy/acp-spec' --api-bridge-url 'http://localhost:8080'
+```
+
+# Running the application with wfsm
+
+Install the workflow server manager:
+<https://docs.agntcy.org/pages/agws/workflow_server_manager.html#installation>
+
+Deploy the application thanks to the wfsm and the manifest file:
+
+```
+wfsm deploy --manifestPath ./deploy/api_bridge_mcp_demo.json
+```
+
+This will build and run the application inside a docker container, and show a
+similar output to the following:
+
+```
+[...]
+
+2025-04-24T17:52:50+02:00 INF ---------------------------------------------------------------------
+2025-04-24T17:52:50+02:00 INF ACP agent deployment name: org.agntcy.api_bridge_agent_demo_mcp
+2025-04-24T17:52:50+02:00 INF ACP agent running in container: org.agntcy.api_bridge_agent_demo_mcp, listening for ACP requests on: http://127.0.0.1:56587
+2025-04-24T17:52:50+02:00 INF Agent ID: d7a2714c-b6bb-4248-8144-ec9c57b0e3ea
+2025-04-24T17:52:50+02:00 INF API Key: 94445385-030e-4e09-9e8c-7631bfdca434
```

Comment:
Can we change the key to something that doesn't look like a key so the secret scanners don't get confused?